src/dmd/dsymbolsem.d

```diff
@@ -1439,7 +1439,9 @@ private extern(C++) final class DsymbolSemanticVisitor : Visitor
                 {
                     uint errors = global.errors;
                     dsym.inuse++;
-                    if (ei)
+                    // Bug 20549. Don't try this on modules or packages, syntaxCopy
+                    // could crash (inf. recursion) on a mod/pkg referencing itself
+                    if (ei && (ei.exp.op != TOK.scope_ ? true : !(cast(ScopeExp)ei.exp).sds.isPackage()))
```

Comment:
`||` is a thing too

```
ei.exp.op != TOK.scope_ || !(cast(ScopeExp)ei.exp).sds.isPackage())
```

Reply:
My initial version was a parenthesesless with the ternary, for some reason I put the parens before committing, but yeah with them that alternative is absolutely more clean.